src/widget/weffectpushbutton.cpp

```diff
@@ -64,7 +64,7 @@ void WEffectPushButton::mousePressEvent(QMouseEvent* e) {
     // The push handler may have set the left value. Check the corresponding
     // QAction.
     double leftValue = getControlParameterLeft();
-    foreach (QAction* action, m_pButtonMenu->actions()) {
+    for (const auto& action : m_pButtonMenu->actions()) {
```

Comment:
so this is a constant reference to a pointer?  That seems poor, "for (auto* action :" would be the preferred way to do it I think.